File: Mixter/Domain/Messages/MessageDeleted.cs
namespace Mixter.Domain.Messages
{
    public struct MessageDeleted : IDomainEvent
    {
        public MessageId MessageId { get; private set; }

        public MessageDeleted(MessageId messageId) : this()
        {
            MessageId = messageId;
        }

        public object GetAggregateId()
        {
            return MessageId;
        }
    }
}

File: app/Domain/Messages/MessageDeleted.php
<?php

namespace App\Domain\Messages;

use App\Domain\Identity\UserId;
use App\Domain\IDomainEvent;

class MessageDeleted implements IDomainEvent
{
    private $messageId;
    private $authorId;

    public function __construct(MessageId $messageId, UserId $authorId)
    {
        $this->messageId = $messageId;
        $this->authorId = $authorId;
    }

    public function getAggregateId(): string
    {
        return $this->messageId->getId();
    }

    public function getMessageId(): MessageId
    {
        return $this->messageId;
    }

    public function getDeleterId(): UserId
    {
        return $this->authorId;
    }
}
